### PR TITLE
Lu 5450

### PR DESCRIPTION
**Pull request by kingmushroom**
Pipeline and tweaks needed for it to work.

Installing dependencies, quality-checks, and deploy, for each of the 3 lambda in the repo.
Terraform apply builds its infrastructure, then builds and deploys deployment ecr image to the newly created ecr repo.

### CODE REVIEW

File: aws-infrastructure/serverless-config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
 "bucketname": "takeon.validation.sandbox.deployment",
-"ingest"  : "takeon-ingest-pck-to-json",
-"ingestcombine"  : "takeon-combine-json-ingest",
-"ingestbricks" : "takeon-ingest-bricks-pck"
-}
\ No newline at end of file
+"ingest"  : "takeon-ingest-pck-to-json-mc-dl",
+"ingestcombine"  : "takeon-combine-json-ingest-mc-dl",
+"ingestbricks" : "takeon-ingest-bricks-pck-mc-dl"
+}
```
Comment (sufyaniqbal) [inline aws-infrastructure/serverless-config.json:6]: possibly replace/remove 'mc-dl'?

File: aws-infrastructure/deployment_ecr.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+resource "aws_ecr_repository" "takeon-cicd" {
+  name                 = "takeon-cicd"
+  image_tag_mutability = "MUTABLE"
+
+  tags = {
```
Comment (RyanAldy) [inline aws-infrastructure/deployment_ecr.tf:5]: Add "app" : "takeon" tag maybe?

File: aws-infrastructure/serverless-config.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 "bucketname": "takeon.validation.sandbox.deployment",
-"ingest"  : "takeon-ingest-pck-to-json",
-"ingestcombine"  : "takeon-combine-json-ingest",
-"ingestbricks" : "takeon-ingest-bricks-pck"
-}
\ No newline at end of file
+"ingest"  : "takeon-ingest-pck-to-json-mc-dl",
```
Comment (RyanAldy) [inline aws-infrastructure/serverless-config.json:3]: Parameterise these to include user?

File: pipeline_params.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+aws_region: eu-west-2
+branch: dev
+environment_name: takeon-dev-sandbox
+environment_name_vpc: takeon-dev-sandbox-vpc
+#user running terraform
+user: mc-dl
```
Comment (RyanAldy) [inline pipeline_params.yml:6]: Remove the user value

File: pipeline_params.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+#STAGE can be replaced with dev, test, prod
+stage: dev
+state_bucket_dynamodb_tab: takeon-terraform-state
+#USER can be replaced with user running terraform
+ingest_state_file: ingest/dev/mc-dl/terraform.tfstate
```
Comment (RyanAldy) [inline pipeline_params.yml:11]: substitute mc-dl with <user> maybe?

**Pull request merged**